From 7d7e1d66ebe23c11bd1084c9c82ced529184b190 Mon Sep 17 00:00:00 2001
From: Felipe Balbi <balbi@ti.com>
Date: Tue, 26 May 2015 14:52:56 -0500
Subject: [PATCH 16/18] remoteproc/pruss: fix pruss_init return on error

If the pru_rproc_driver fails to register in pruss_init(),
the failure code should be returned instead of returning 0,
a success value. So, fix the pruss_init() return code
accordingly.

Signed-off-by: Felipe Balbi <balbi@ti.com>
Signed-off-by: Suman Anna <s-anna@ti.com>
---
 drivers/remoteproc/pruss_remoteproc.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/remoteproc/pruss_remoteproc.c b/drivers/remoteproc/pruss_remoteproc.c
index b567b7a..3e374a0 100644
--- a/drivers/remoteproc/pruss_remoteproc.c
+++ b/drivers/remoteproc/pruss_remoteproc.c
@@ -1371,7 +1371,7 @@ static int __init pruss_init(void)
 		platform_driver_unregister(&pruss_driver);
 	}
 
-	return 0;
+	return ret;
 }
 module_init(pruss_init);
 
-- 
1.7.9.5

